ql/src/java/org/apache/hadoop/hive/ql/reexec/ReOptimizePlugin.java

```diff
@@ -80,14 +83,15 @@ public void initialize(Driver driver) {
     coreDriver.getHookRunner().addOnFailureHook(statsReaderHook);
     coreDriver.getHookRunner().addPostHook(statsReaderHook);
     alwaysCollectStats = driver.getConf().getBoolVar(ConfVars.HIVE_QUERY_REEXECUTION_ALWAYS_COLLECT_OPERATOR_STATS);
+    maxExecutions = 1 + coreDriver.getConf().getIntVar(ConfVars.HIVE_QUERY_MAX_REEXECUTION_COUNT);
```

Comment:
could we move this under IReExecutionPlugin default method? and override shouldReExecute when needed?